src/org/openjdk/asmtools/jcoder/Main.java

```diff
@@ -90,6 +114,12 @@ public void usage() {
         environment.info("info.opt.version");
     }
 
+    // Run jcoder compiler with args
+    public synchronized boolean compile(String... argv) {
```

Comment:
I like the compile method with args idea! Why it was done sudenly synchrornized! Whole asmtools are not thread safe, aren't they?